### ISSUE TITLE
Question on Normalization

### ISSUE THREAD

**Issue by ssram50** [lucidrains/lambda-networks#24]
Why use BatchNorm2 and not softmax for the keys normalization?
Why normalize both keys and values with BatchNorm2?

Original paper states:
"Generating the contextual lambda function. Our lambda layer first computes keys and values
by linearly projecting the context, and keys are normalized across context positions via a softmax
operation yielding normalized keys.
.

**Comment by vackosar** [comment]
So what is the question?

**Comment by johndpope** [comment]
close
